netbox/extras/jobs.py

```diff
@@ -42,10 +44,18 @@ def run_script(self, script, request, data, commit):
                 # A script can modify multiple models so need to do an atomic lock on
                 # both the default database (for non ChangeLogged models) and potentially
                 # any other database (for ChangeLogged models)
-                with transaction.atomic():
-                    script.output = script.run(data, commit)
-                    if not commit:
-                        raise AbortTransaction()
+                branch_db = router.db_for_write(Device)
+                with transaction.atomic(using='default'):
+                    # If branch database is different from default, wrap in a second atomic transaction
+                    if branch_db != 'default':
```

Comment:
Can we make this more generic, rather than talking specifically about a `branch_db`?